app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,48 @@
+class PhotosController < ApplicationController
+  def index
+    @photos = Photo.all.order(created_at: :desc)
+  end
+
+  def show
+    @photo = Photo.find(params[:id])
+    @average = @photo.avg
+  end
+
+  def new
+    @photo = Photo.new
+  end
+
+  def create
+    @photo = Photo.new(photo_params)
```

Comment:
You could also put .merge(user: current_user) on the end of your require/permit call in photo_params